### PR TITLE
Jenkins - notify Slack

### LINKED ISSUE #165: Set up Jenkins to send error message to Slack
**Is your feature request related to a problem? Please describe.**
When the tests fail, send a slack message to #psammead.

**Describe the solution you'd like**

**Describe alternatives you've considered**
We could dive into setting up _everything_ we want on Jenkins right away, but let's make the chunks more manageable.

**Testing notes**
[Tester to complete]

Dev insight: open a PR with a deliberately failing test - should see Slack message along the lines of "PR was merged into Psammead".

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Resolves #165

Notifies Slack of build fail/pass/succeed.

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval

### CODE REVIEW

File: Jenkinsfile
Comment (sareh) [inline Jenkinsfile]: It would be better to pass the channel as a parameter to this function, so that we can re-use this function in future / pull it out of Psammead into a shared library.

Comment (sareh) [reply Jenkinsfile]: Also, are there docs for the global methods available in Jenkins? Where can I inspect them?

File: Jenkinsfile
Comment (sareh) [inline Jenkinsfile]: Could this condition be simplified to be `if (GIT_BRANCH == 'latest')`? Same with the others below.

**Review by sareh — CHANGES_REQUESTED** [review]
I've added some comments about refactoring. 
I'd like for this to be tested before being merged - perhaps the best way to do that would be to change the condition to work on this branch temporarily, see the slack notification work, so we can add a screenshot to the PR, then to restrict it to `latest` again.

I'm open to hearing other suggestions for testing it, though!

File: Jenkinsfile
Comment (ChrisBAshton) [reply Jenkinsfile]: There are methods documented but can't put the links in a public repo :) I'll add to the infrastructure README

File: Jenkinsfile
Comment (ChrisBAshton) [reply Jenkinsfile]: Not sure - I think it's like this for a reason. (It was copied & pasted from Simorgh).
The 'success' conditions are different, that one just checks `params.BRANCH` (not `GIT_BRANCH`).

It's something to do with not spamming the Slack channel, but I don't know exactly what it does. 🤔 Worth documenting once we figure it out!

Comment (sareh) [reply Jenkinsfile]: `params.BRANCH` is chosen in the Jenkins Params interface. 

We have that defined in the Simorgh Infra Jenkins pipeline (https://github.com/BBC/simorgh-infrastructure/blob/latest/Jenkinsfile#L103-L104)  but we don't have this defined in this Psammead pipeline, so it can be removed.

**Comment by ChrisBAshton** [comment]
Blocked on #188

**Comment by bcmn** [comment]
No longer blocked on #188.

File: Jenkinsfile
Comment (ChrisBAshton) [inline Jenkinsfile:10]: 👍

File: Jenkinsfile
Comment (ChrisBAshton) [inline Jenkinsfile:84]: Can we refactor the steps below?

```suggestion
post {
  when {
    GIT_BRANCH == 'latest'
   }
 }
```

Disclaimer: haven't tested this, but [should work in theory](https://jenkins.io/doc/book/pipeline/syntax/#when).

**Review by ChrisBAshton — COMMENTED** [review]
Looks good - one suggestion for refactor

File: Jenkinsfile
Comment (dr3) [reply Jenkinsfile:84]: Alas no, sadly it says 
```
WorkflowScript: 60: Invalid condition "when" - valid conditions are [always, changed, fixed, regression, aborted, success, unstable, failure, notBuilt, cleanup] @ line 60, column 5.

       when {
```
Re: https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/psammead/detail/jenkins/8/pipeline

Comment (ChrisBAshton) [reply Jenkinsfile:84]: Shame! The only other thought is to refactor with closures:

```
def onLatestOnly(callback) {
  if (GIT_BRANCH == 'latest') {
    callback()
  }
}
```

Then lower down we'd:

```
onLatestOnly(notifySlack('danger', 'Aborted', gitCommitAuthor, stageName, gitCommitMessage, slackChannel))
```

Comment (ChrisBAshton) [reply Jenkinsfile:84]: Would be useful on line 43 too!

**Review by sareh — APPROVED** [review]
👍 Looks great. I branched off this branch `jenkins-test` and changed the 'latest' branch to 'jenkins-test'. It succeeded: 
screen shot jenkins
https://ci.news.tools.bbc.co.uk/job/psammead/job/jenkins-test/2/consoleFull

**Review by bcmn — APPROVED** [review]

**Pull request merged**
